### PR TITLE
Avoid global mutation.

### LINKED ISSUE #851: Inadvertent global mixin
Looking at how things are using `FileMixin` I ran across https://github.com/chef/ohai/blob/80c91e74ac403959d54c9f15b1aa7a3c4fc4bd8b/lib/ohai/plugins/linux/sessions.rb#L21

I think this line is in error, it is mixing the helper in globally (i.e. in to `Object`). I think the intention was to do it inside the plugin body but when things were moved from the old DSL to the new one, it got mixed up. This should be removed ASAP because dumping things into the global namespace is terrifying, but doing this backwards compat might be tricky. My vote would be to add that mixin the `Plugin` base class so any existing plugins will continue to work.

### PR DESCRIPTION
**Pull request by coderanger**
Closes #851.

### CODE REVIEW

File: lib/ohai/dsl/plugin.rb [hunk 1]
```diff
@@ -82,6 +83,10 @@ class Plugin
       include Ohai::Mixin::Command
       include Ohai::Mixin::SecondsToHuman
 
+      # Pending next major release of Ohai.
```
Comment (tas50) [inline lib/ohai/dsl/plugin.rb]: Removal pending? The comment is a bit vague. Also when we leave comments in code that say next major version no action is ever taken. Pick the version where we plan to deprecate it.

Comment (coderanger) [inline lib/ohai/dsl/plugin.rb]: We should probably decide first if we want it to be removed at all or if FileHelper should just stay as part of the Plugin base API. That would seem fine too :)

Comment (tas50) [inline lib/ohai/dsl/plugin.rb]: I'd be fine if we kept this part of the Plugin base API. It's small and useful for writing plugins.

Comment (coderanger) [inline lib/ohai/dsl/plugin.rb]: Roger, updating PR :)

**Comment by coderanger** [comment]
The DmiDecode change _might_ break something but it seems much less likely someone was accidentally using those methods :) Will see what the tests say.

**Comment by tas50** [comment]
I highly doubt anyone is using the DMI helper and it's the right thing to do. Lets just fix that.

**Comment by tas50** [comment]
+1

**Pull request merged**